src/main/java/com/google/firebase/auth/FirebaseUserManager.java

```diff
@@ -230,20 +230,28 @@ UserImportResult importUsers(UserImportRequest request) throws FirebaseAuthExcep
 
   Tenant getTenant(String tenantId) throws FirebaseAuthException {
     GenericUrl url = new GenericUrl(tenantMgtBaseUrl + "/tenants/" + tenantId);
-    Tenant response = sendRequest("GET", url, null, Tenant.class);
-    if (Strings.isNullOrEmpty(response.getTenantId())) {
-      throw new FirebaseAuthException(TENANT_NOT_FOUND_ERROR, "Failed to get tenant.");
-    }
-    return response;
+    return sendRequest("GET", url, null, Tenant.class);
+  }
+
+  Tenant createTenant(Tenant.CreateRequest request) throws FirebaseAuthException {
+    GenericUrl url = new GenericUrl(tenantMgtBaseUrl + "/tenants");
+    url.putAll(request.getProperties());
+    return sendRequest("POST", url, null, Tenant.class);
+  }
+
+  Tenant updateTenant(Tenant.UpdateRequest request) throws FirebaseAuthException {
+    Map<String, Object> properties = request.getProperties();
+    checkArgument(properties.size() > 1,
+        "tenant update must have at least one property other than tenantId set");
+    GenericUrl url = new GenericUrl(tenantMgtBaseUrl + "/tenants/" + request.getTenantId());
```

Comment:
You need to generate an `updateMask` from the given properties, and add it the URL here. See the corresponding Go and Node.js implementations.

Reply:
Okay, interesting. So for the update request, we encode the properties 2 different ways.

I found the utility which builds the mask in Node (https://github.com/firebase/firebase-admin-node/blob/0f6c02e3377c3337e4f206e176b2d96ec6dd6c3c/src/utils/index.ts#L148).

Any idea if we already have a similar utility in Java? Or do I need to write one? From looking at Node's tests (https://github.com/firebase/firebase-admin-node/blob/0f6c02e3377c3337e4f206e176b2d96ec6dd6c3c/test/unit/utils/index.spec.ts#L307), it looks like the format we want is something like this:

...../tenants/TENANT_ID?updateMask=name.TENANT_ID,displayName.DISPLAY_NAME,allowPasswordSignup.true,enableEmailLinkSignin.false

Does that look right?

Reply:
I don't think we have a util in Java for this at the moment. The field mask should only consist of field names in a dotted notation. Following json payload:

```
{
  'parentField1': 'some value',
  'parentField2': 'another value',
  'parentField3': {
    'childField1': 'yet another value'
  }
}
```

should become:
```
['parentField1', 'parentField2', 'parentField3.childField1']
```

You can also look at the unit tests of the Node.js utils for more examples.

Reply:
Done.